lib/cellular/backends/sendega.rb

```diff
@@ -35,49 +35,72 @@ def self.deliver(options = {}, savon_options = {})
         # http://controlpanel.sendega.no/Content/Sendega%20-%20API%20documentation%20v2.3.pdf
 
         savon_options[:wsdl] = GATEWAY_URL
-
+        request_queue = {}
         client = Savon.client savon_options
 
-        result = client.call(:send, message: {
-            username: Cellular.config.username,
-            password: Cellular.config.password,
-            sender: options[:sender],
-            destination: options[:recipient],
-            pricegroup: options[:price] || 0, # default price to 0
-            contentTypeID: 1,
-            contentHeader: '',
-            content: options[:message],
-            dlrUrl: Cellular.config.delivery_url,
-            ageLimit: 0,
-            extID: '',
-            sendDate: '',
-            refID: '',
-            priority: 0,
-            gwID: 0,
-            pid: 0,
-            dcs: 0
-          }
-        )
-
-        body = result.body[:send_response][:send_result]
+        receipients_batch(options).each_with_index do |batch, index|
+          result = client.call(:send, message: payload(options, batch))
 
-        if body[:success]
-          [
-            body[:error_number].to_i,
-            'Message is received and is being processed.'
-          ]
-        else
-          [
-            body[:error_number].to_i,
-            body[:error_message]
-          ]
+          request_queue[index] = {
+            batch: batch,
+            result: result,
+            body:result.body[:send_response][:send_result],
+            response: map_response(result.body[:send_response][:send_result])
+          }
         end
+
+        # for now just resturn first response
+        request_queue[0][:response]
       end
 
       def self.receive(data)
         raise NotImplementedError
       end
 
+      def self.savon_config
+        {
+           username: Cellular.config.username,
+           password: Cellular.config.password,
+           dlrUrl: Cellular.config.delivery_url
+        }
+      end
+
+      def self.payload(options, recipients)
+       {
+          sender: options[:sender],
+          destination: recipients,
+          pricegroup: options[:price] || 0, # default price to 0
+          contentTypeID: 1,
+          contentHeader: '',
+          content: options[:message],
+          ageLimit: 0,
+          extID: '',
+          sendDate: '',
+          refID: '',
+          priority: 0,
+          gwID: 0,
+          pid: 0,
+          dcs: 0
+        }.merge!(savon_config)
+      end
```

Comment:
Then again I'm pretty pedantic about method names and keeping their arguments to a minimum.